docs/source/tutorials/memory_optimizations.rst

```diff
@@ -208,10 +213,56 @@ To understand how this works, we encourage you to read through the relevant PyTo
-single-device full finetune recipe. This feature works best when gradient memory is particularly large;
+single-device full finetune recipe. This feature works best when optimizer memory is particularly large;
 e.g. when using a stateful optimizer with a model with a lot of parameters, and when you don't need to use
 :ref:`gradient accumulation <glossary_grad_accm>`.
 
+.. _glossary_cpu_offload:
+
+Offloading Optimizer/Gradient states to CPU
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+*What's going on here?*
+
+We've mentioned above the concept of optimizer states - memory used by the stateful optimizers to maintain a state of gradient statistics, and
+model gradients - tensors used to store gradients when we perform model backwards passes. We support using CPU offloading in our single-device recipes
+through the `CPUOffloadOptimizer <https://github.com/pytorch/ao/tree/main/torchao/prototype/low_bit_optim#optimizer-cpu-offload>`_ from ``torchao``.
+
+This optimizer can wrap any base optimizer and works by keeping the optimizer states and performing the optimizer step on CPU, thus reducing
+GPU memory usage by the size of the optimizer states. Additionally, we can also offload gradients to the CPU by using `offload_gradients=True`.
+
+*Sounds great! How do I use it?*
+
+To use this optimizer in your recipes, set the ``optimizer`` key in your config to :class:`torchao.prototype.low_bit_optim.CPUOffloadOptimizer`, which
+will use the :class:`torch.optim.AdamW` optimizer with ``fused=True`` as the base optimizer. For example, to use this optimizer to offload
+both optimizer states and gradients to CPU:
+
+.. code-block:: bash
+
+  tune run <RECIPE> --config <CONFIG> \
+  optimizer=optimizer=torchao.prototype.low_bit_optim.CPUOffloadOptimizer \
+  optimizer.offload_gradients=True \
+  lr=4e-5
+
+
+or by directly :ref:`modifying a config file<config_tutorial_label>`:
+
+.. code-block:: yaml
+
+  optimizer:
+    _component_: torchao.prototype.low_bit_optim.CPUOffloadOptimizer
+    offload_gradients: True
+    # additional key-word arguments can be passed to torch.optim.AdamW
+    lr: 4e-5
+
+Some helpful hints from the ``torchao`` `CPUOffloadOptimizer page <https://github.com/pytorch/ao/tree/main/torchao/prototype/low_bit_optim#optimizer-cpu-offload>`_:
+
+* The CPU optimizer step is often the bottleneck when optimizer CPU offload is used. To minimize the slowdown, it is recommended to (1) use full ``bf16`` training so that parameters, gradients, and optimizer states are in ``bf16``; and (2) give GPU more work per optimizer step (e.g. larger batch size with activation checkpointing, gradient accumulation).
+* Gradient accumulation should always be set to 1 when ``offload_gradients=True``, as gradients are cleared on GPU every backward pass.
+* This optimizer works by keeping a copy of parameters and pre-allocating gradient memory on CPU. Therefore, expect your RAM usage to increase by 4x model size.
```

Comment:
Is it exactly 4x model size? Or roughly?

Reply:
According to the ao docs yeah 

> To minimize the amount of CPU<->GPU data transfer, we keep a copy of parameters and pre-allocate gradients memory on CPU. Therefore, expect your RAM usage to increase by 2x model size + optimizer state (which is 2x model size for Adam).

and since we always use adam it is 4x